audioipc/src/rpccore.rs

```diff
@@ -58,48 +59,44 @@ type ProxyRequest<Request> = (ProxyKey, Request);
 // Each Proxy is registered with the ClientHandler on initialization via the
 // ProxyManager and unregistered when dropped.
 // A ClientHandler normally lives until the last Proxy is dropped, but if the ClientHandler
-// encounters an internal error, `response_tx` will be closed and `proxy_mgr` can
-// no longer be upgraded to register new Proxies.
+// encounters an internal error, `handler_tx` and `response_tx` will be closed.
 #[derive(Debug)]
 pub struct Proxy<Request, Response> {
     handle: Option<(EventLoopHandle, Token)>,
-    key: ProxyKey,
+    key: Result<ProxyKey>,
```

Comment:
It seems weird to me to have a `Result` typed member. Could `new` return `Result<Self>` so it can return `Err` when `proxy_mgr.register_proxy(...)` fails?

Reply:
Done.